proxy/jsonproto.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+)
+
+type Request struct {
+	Id      uint             `json:"id"`
+	Command string           `json:"command"`
+	Data    *json.RawMessage `json:"data"`
+}
+
+type Response struct {
+	Id      uint                   `json:"id"`
+	Success bool                   `json:"success"`
+	Error   string                 `json:"error,omitempty"`
+	Data    map[string]interface{} `json:"data,omitempty"`
+}
+
+// XXX: could do with its own package to remove that ugly namespacing
```

Comment:
Indeed :) And it could be called protocol.go. The JSON formatting is only an implementation detail.